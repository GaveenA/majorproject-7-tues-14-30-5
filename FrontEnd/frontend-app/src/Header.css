a {
    font-size: 1rem;
    padding: 0.5rem;
    margin: 0.5rem;
}

a.current {
    text-decoration: overline underline;
}