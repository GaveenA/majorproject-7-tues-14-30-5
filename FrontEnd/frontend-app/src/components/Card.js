import { Button } from "@material-ui/core";
import React, { useState } from "react";
import "./Card.css";
import Popup from './Popup.js';


  
    // <Button variant="primary" onClick={() => setShow(true)}>
    //       Custom Width Modal
    //     </Button>


function Card({ src, title, description, price }) {

  // const [value, setPopup] = useState(false);

  // const updateState = (value) => {
  //   setPopup(value)
  //   console.log("Changed State: value = ",value);
  // }


  const [isClick, setIsClick] = useState(false);




  console.log("value at 26 = ",isClick);
  return (
    <div className="card">
          <img src={src} alt="" />
          <div className="card__info">
            <h2>{title}</h2>
            <h4>{description}</h4>
            <h3>{price}</h3>
          </div>
    </div>
  );
}

export default Card;
