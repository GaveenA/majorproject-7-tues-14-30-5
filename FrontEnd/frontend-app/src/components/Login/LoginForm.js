import React, { useState } from 'react'
import {useSelector, useDispatch} from 'react-redux'
import { useHistory } from 'react-router-dom'
import { login } from '../../redux'
import { Button, Form } from "react-bootstrap";
import './loginForm.css'
import Logo from '../../public/logo/Logo.png'


function LoginForm() {

    const [customerLogin, setLogin] = useState({
        username:'',
        password:''
    })
    
    const history = useHistory()

    const loggedIn = useSelector(state => state.login.loggedIn)
    const dispatch = useDispatch()

    if (loggedIn) {
        history.push('/')
    }

    return (
        <div className='login'>
          <div className='login__container'>
            <img className='loginBox__logo' src={Logo} alt='AGME_Logo' />
            <Form.Row id="username">
                <input type='text' value={customerLogin.username} onChange={e => setLogin({...customerLogin, username: e.target.value})} placeholder="Username" />
            </Form.Row>
            <Form.Row id="password">
                <input type='password' value={customerLogin.password} onChange={e => setLogin({...customerLogin, password: e.target.value})} placeholder="Password" />
                </Form.Row>
                <Form.Row id="submit_button">
                <input type='button' onClick={()=>dispatch(login(customerLogin))} value='Login'/>
                </Form.Row>
          </div>

        </div>

    )

}

export default LoginForm