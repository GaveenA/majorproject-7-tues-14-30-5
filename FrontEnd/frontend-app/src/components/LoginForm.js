import React, {Component} from "react";
import { Button, Form } from 'react-bootstrap';
import axios from 'axios';

export default class LoginForm extends Component {

    constructor(props) {
        super(props)
        this.state = {
		username:"",
		password:""
	};
        this.signIn = this.signIn.bind(this)
        this.formChange = this.formChange.bind(this)
    }
    signIn(event)
    {
        const data = {
		username: this.state.username,
		password: this.state.password
	};
        axios.post('http://localhost:8080/login', data).then(r => console.log(r)).catch(error => console.log(error));
        event.preventDefault()
    }
    formChange(event)
    {
        this.setState({
            [event.target.name]:event.target.value
        });
    }


    render() {
        return (
            <div className="col-md-5 col-md-offset-2">
                <Form onSubmit={this.signIn} id="loginFormId">
                    <Form.Group controlId="formBasicEmail">
                        <Form.Row>
                            <Form.Label>Username</Form.Label>
                            <Form.Control required name="username" value={this.state.username} onChange={this.formChange} type="username" placeholder="Username" />
                        </Form.Row>
                    </Form.Group>

                    <Form.Group controlId="formBasicPassword">
                        <Form.Row>
                            <Form.Label>Password</Form.Label>
                            <Form.Control required name="password" value={this.state.password} onChange={this.formChange} type="password" placeholder="Password" />
                        </Form.Row>
                    </Form.Group>

                    <Button variant="success" type="submit">
                        Submit
                    </Button>
                </Form>
            </div>
        )

    }

}
