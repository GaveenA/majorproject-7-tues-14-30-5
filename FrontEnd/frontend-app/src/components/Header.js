import React, {Component} from "react";
import { Button, Form } from 'react-bootstrap';
import axios from 'axios';
import './Header.css'
import Logo from "../Logo.png";

import LanguageIcon from "@material-ui/icons/Language";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import { Avatar } from "@material-ui/core";

import { Link } from "react-router-dom";

import Navbar from "./Navbar"

export default class LoginForm extends Component {

    constructor(props) {
        super(props)
        this.state = {
		username:"",
		password:""
	};
        this.signIn = this.signIn.bind(this)
        this.formChange = this.formChange.bind(this)
    }
    signIn(event)
    {
        const data = {
		username: this.state.username,
		password: this.state.password
	};
        axios.post('http://localhost:8080/login', data).then(r => console.log(r)).catch(error => console.log(error));
        event.preventDefault()
        
    }
    formChange(event)
    {
        this.setState({
            [event.target.name]:event.target.value
        });
    }

 
    render() {
        return (
          <div className="header">

          <Navbar/>

            <Link to='/' className= "header__icon">
                <img
                className="header__icon"
                src={Logo}
                alt=""
                />
            </Link>
              
            <div className="header__center">
              <button class="home__button">HOME</button>
              <button class="About__button">ABOUT</button>
              <button class="Contact__button">CONTACT</button>
            </div>


          </div>
        );

    }

}
