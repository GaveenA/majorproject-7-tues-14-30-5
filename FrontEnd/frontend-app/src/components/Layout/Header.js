import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { logout } from '../../redux'
import { Link, useHistory } from 'react-router-dom'

import './Header.css'
import Logo from '../../public/logo/Logo.png'
import Navbar from './Navbar'

function Header() {
    
    const loggedIn = useSelector(state => state.login.loggedIn)
    const dispatch = useDispatch()

    const history = useHistory()
    
    return (
        <div className="header">    
            <Navbar/>
            <Link to='/' className= "header__icon">
                <img className="header__icon" src={Logo} alt="AGME_Logo" />
            </Link>
              
            <div className="header__center">
                <button className="home__button" onClick={() => history.push('/')}>HOME</button>
                {!loggedIn ? 
                    <button className="About__button" onClick={() => history.push('/login')}>LOGIN</button> :
                    <button className="About__button" onClick={() => dispatch(logout)}>LOGOUT</button> 
                }
                
                <button className="Contact__button" onClick={() => history.push('/contact')} >CONTACT</button>
                <button className="Abt__button" onClick={() => history.push('/about')} >ABOUT</button>
                <button className="regist__button" onClick={() => history.push('/registration')} >SIGN UP</button>


            </div>

      
        </div>


        
        
        // if (!loggedIn) {
            
            //     <Navbar bg="dark" variant="dark">
            //         <Navbar.Brand href="/">AGME</Navbar.Brand>
        //         <Nav className="mr-auto">
        //             <Nav.Link href="/">Home</Nav.Link>
        //             <Nav.Link href="/about">About</Nav.Link>
        //             <Nav.Link href="/contact">Contact Us</Nav.Link>
        //         </Nav>
        //         <Nav className="ml-auto">
        //             <Nav.Link href="/login" activeclassname="current">Log In</Nav.Link>
        //             <Nav.Link href="/registration">Sign Up</Nav.Link>
        //         </Nav>
        //     </Navbar>
            
        // } else {
        //     return (
        //         <Navbar bg="dark" variant="dark">
        //         <Navbar.Brand href="/">AGME</Navbar.Brand>
        //         <Nav className="mr-auto">
        //         <Nav.Link href="/">Home</Nav.Link>
        //         <Nav.Link href="/about">About</Nav.Link>
        //         <Nav.Link href="/contact">Contact Us</Nav.Link>
        //         </Nav>
        //         <Nav className="ml-auto">
        //         <Nav.Link href="/" activeclassname="current" onClick={() => dispatch(logout)}>Log Out</Nav.Link>
        //         </Nav>
        //         </Navbar>
        //     )
        // }
    )
}

export default Header