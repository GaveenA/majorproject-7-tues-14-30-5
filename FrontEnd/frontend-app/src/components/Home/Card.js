import React from "react";
import "./Card.css";
  
    // <Button variant="primary" onClick={() => setShow(true)}>
    //       Custom Width Modal
    //     </Button>


function Card({ src, title, description, price }) {

  // const [value, setPopup] = useState(false);

  // const updateState = (value) => {
  //   setPopup(value)
  //   console.log("Changed State: value = ",value);
  // }


  // const [isClick, setIsClick] = useState(false);


  // console.log("value at 26 = ",isClick);
  return (
    <div className="card" data-testid = "card-test">
          <img src={src} alt="" />
          <div className="card__info">
            <h2>{title}</h2>
            <h4>{description}</h4>
            <h3>{price}</h3>
          </div>
    </div>
  );
}

export default Card;
