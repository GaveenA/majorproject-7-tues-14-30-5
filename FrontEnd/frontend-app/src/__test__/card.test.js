import React from 'react';
 import ReactDOM from 'react-dom';
 import Card from '../components/Home/Card';
 import Header from '../components/Login/LoginForm.js';

import {render, cleanup, getByTestId} from "@testing-library/react";
import legal from "../public/images/legal.jpg";
import "@testing-library/jest-dom" ;
import LoginForm from '../components/Login/LoginForm.js';

const ReactTestRenderer = require('react-test-renderer');

afterEach(cleanup)

 it ("renders without crashing", () =>   {
    const div = document.createElement("div");
    expect(ReactDOM.render(<Card></Card>,div)).toBeTruthy;
 });


 it("renders card correctly", ()=> {
     render(<Card
        src={legal}
        title="Pearson Spector - Law Firm"
        description="Expert Legal services, led by a team of exceptional Attorneys."
        />)
        // expect(getByTestId('card-test')).toHaveProperty()
        // expect(getByTestId('card-test')).toHaveAttribute()
        // const object = getByTestId(document, 'card-test');
        
        expect(getByTestId(document, 'card-test')).toHaveProperty('title')
        expect(getByTestId(document, 'card-test')).toHaveTextContent("Pearson Spector - Law Firm")
        expect(getByTestId(document, 'card-test')).toHaveTextContent("Expert Legal services, led by a team of exceptional Attorneys.")
        
 });


 it("matches-snapshot", () => {
    const tree = ReactTestRenderer.create(<Card
        src={legal}
        title="Pearson Spector - Law Firm"
        description="Expert Legal services, led by a team of exceptional Attorneys."
        />).toJSON();

    expect(tree).toMatchSnapshot();
 })



