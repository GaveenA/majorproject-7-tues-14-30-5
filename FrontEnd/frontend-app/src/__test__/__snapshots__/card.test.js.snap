// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches-snapshot 1`] = `
<div
  className="card"
  data-testid="card-test"
>
  <img
    alt=""
    src="legal.jpg"
  />
  <div
    className="card__info"
  >
    <h2>
      Pearson Spector - Law Firm
    </h2>
    <h4>
      Expert Legal services, led by a team of exceptional Attorneys.
    </h4>
    <h3 />
  </div>
</div>
`;
